src/plugins/Plugin.js

```diff
@@ -89,6 +89,18 @@ module.exports = class Plugin {
     }
   }
 
+  setMetaFromTargetForm () {
+    const el = findDOMElement(this.opts.target)
+    const formMetaData = getMetaFromForm(el)
+    if (formMetaData) {
+      this.core.log('Adding metadata from form:')
+      this.core.log(formMetaData)
+      this.core.setMeta(formMetaData)
+    } else {
+      this.core.log('Couldn’t extract metadata from form')
+    }
+  }
```

Comment:
Yeah, I thought that this feels out of place too, your suggested solution is perfect.